src/plugins/explorer/sync-transactions.js

```diff
@@ -88,7 +91,41 @@ function createSyncer (config, eventBus, web3, queue, eventsRegistry, indexer) {
       })
   }
 
-  const getPastEvents = (fromBlock, toBlock, address) =>
+  function getPastEventsWithChunks (options) {
+    const CHUNK_SIZE = 4000
+    const { address, contract, eventName, fromBlock, toBlock, filter, metaParser, minBlock = 0, onProgress = noop } = options
+    let chunkIndex = 0
+    return pWhilst(
+      () => (fromBlock + CHUNK_SIZE * chunkIndex) < toBlock,
+      function () {
+        const newFromBlock = fromBlock + CHUNK_SIZE * chunkIndex
+        const newToBlock = Math.min(fromBlock + CHUNK_SIZE * (chunkIndex + 1), toBlock)
+        debug('Retrieving from %s to %s for event %s', newFromBlock, newToBlock, eventName)
+        return pTimeout(
+          contract
+            .getPastEvents(eventName, {
+              fromBlock: Math.max(newFromBlock, minBlock),
+              toBlock: Math.max(newToBlock, minBlock),
```

Comment:
Seems better to include this lower limit in the `newFromBlock`/`newToBlock` calculations above.

Reply:
Applied here https://github.com/autonomoussoftware/metronome-wallet-core/pull/32/commits/d210679451505871c1f651ee8771624fe4ceb25d